pomerium/gateway/filters.go

```diff
@@ -54,3 +73,54 @@ func applyRedirectFilter(route *pb.Route, filter *gateway_v1.HTTPRequestRedirect
 	}
 	route.Redirect = &rr
 }
+
+func applyExtensionFilter(
+	route *pb.Route,
+	config *model.GatewayConfig,
+	routeConfig *model.GatewayHTTPRouteConfig,
+	filter *gateway_v1.LocalObjectReference,
+) {
+	// Make sure the API group is the one we expect.
+	if filter.Group != gateway_v1.Group(icgv1alpha1.GroupVersion.Group) {
+		return
+	}
+
+	k := model.ExtensionFilterKey{
+		Kind:      string(filter.Kind),
+		Namespace: routeConfig.Namespace,
+		Name:      string(filter.Name),
+	}
+	f := config.ExtensionFilters[k]
+	if f == nil {
```

Comment:
should route referencing a policy that doesn't exit be treated as an error? i.e. it may be referencing an important deny policy